pkg/preparer/podprocess/reporter.go

```diff
@@ -408,7 +413,31 @@ func (r *Reporter) getLastID() (int64, error) {
 		return 0, err
 	}
 
-	return strconv.ParseInt(string(bytes), 10, 64)
+	ret, err := strconv.ParseInt(string(bytes), 10, 64)
+	if err != nil {
+		return 0, util.Errorf("could not parse last processed ID from %q as an int64: %s", r.workspaceFilePath(), err)
+	}
+
+	return ret, nil
+}
+
+// repairLastID() writes the last ID from the sqlite database to the workspace
+// file. It should only be called if the contents of the file are unusable
+// (e.g. 0 byte file or the contents aren't an integer)
+func (r *Reporter) repairLastID() (int64, error) {
+	lastID, err := r.finishService.LastFinishID()
+	if err != nil {
+		r.logger.WithError(err).Errorln("unable to repair workspace file due to error querying for last ID")
+		return 0, err
+	}
+
+	err = r.writeLastID(lastID)
+	if err != nil {
+		r.logger.WithError(err).Errorln("unable to repair workspace file due")
```

Comment:
Perhaps the "due" at the end can be removed, since I'm assuming `WithError` will have given the cause